# the null hypothesis: the % of the former inmates being employed 3 years after the release is the same or lower for participants of the program than the control group, at the significance level of 0.05.

##$H_0 :  P_0 - P_1  \geq 0$

## $H_a : P_0 - P_1 < 0$

## $\alpha = 0.05$

## $P_0 = 0.521, P_1 = 0.533$

In [1]:
import numpy as np

In [2]:
alpha = 0.05
p_0 = 0.521
p_1 = 0.533
n_0 = 409
n_1 = 564

Nt_0 = p_0*n_0
Nt_1 = p_1*n_1

In [3]:
# define sample proportion
sp = (p_0*n_0 + p_1*n_1)/(n_1+n_0)
print sp

0.527955806783


In [4]:
# standard deviaiton funciton
sp_stdev = lambda p, n: np.sqrt(p*(1-p)/n[0] + p*(1-p)/n[1] )
sp_stdev_3y = sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0, n_1])
print p_0, n_0, n_1, sp_stdev_3y

0.521 409 564 0.0324224126866


In [5]:
# z-score function
zscore = lambda p0, p1, s: (p0 - p1)/s
z_3y = zscore(p_1, p_0, sp_stdev_3y)
print z_3y

0.370114343926


## According to the z-score table, P(Z<z)$\approx$0.644

In [6]:
p_3y = 1 - 0.644

In [7]:
if p_3y > alpha:
    print 'the p-value %0.2f is larger than or equal to %0.2f.' % (p_3y, alpha)
    print 'Do not reject the Null Hypothesis!'
else:
    print 'the p-value %0.2f is less than %0.2f.' % (p_3y, alpha)
    print 'Reject the Null Hypothesis!'

the p-value 0.36 is larger than or equal to 0.05.
Do not reject the Null Hypothesis!


## Now for the chi-squared test

## The null hypothesis: the % of the former inmates with postprogram unsubsidized employment for more than 6 weeks is the same or lower for participants of the program than the control group, at the significance level of 0.05.


| Ever employed  | Employed in unsubsidized job | not employed | total number of people |
| -------------  |:---------------------------: | ------------:| ---------------------: |
| test sample    |       14.7\*5.46             | 85.3\*5.64   |        564             |
| control sample |       11.9\*4.09             | 88.1\*4.09   |        409             |
| total          |       115.787                | 857.213      |        973             |

In [8]:
Ntot = 973
expected = 564*409*115.787*857.213
sample_values = [[14.7*5.64, 85.3*5.64], [11.9*4.09, 88.1*4.09]]
chi_stats = lambda N, values, expect: N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chi_stats(Ntot, sample_values, expected)

1.772893977


## The chi-squared test value 3.84 (with 1 degree of freedom) stands for the level of confidence 0.05. Our test value 1.77 gives us a p-value between 0.1 and 0.25. p_row5 $\approx$ 0.22. Since p_row5 > $alpha$, we woundn't want to reject the null hypothesis.